### PR TITLE
xmlWS-3.0: Addition of test for policy skip settings

### LINKED ISSUE #20487: xmlWS-3.0: Addition of test for policy skip settings
This is the test for issue #20439 testing all CXF versions.

### PR DESCRIPTION
**Pull request by BerksanAtes**
Fixes #20487
PropertySettingTests are added to com.ibm.ws.jaxws.2.2.webcontainer_fat and com.ibm.ws.jaxws.X.wsat_fat FAT buckets in order to check property setting functionalities.

### CODE REVIEW

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/FATSuite.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
                 HttpConduitPropertiesTest.class,
                 LoggingTest.class,
                 MTOMTest.class,
+                PropertySettingTest.class,
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/FATSuite.java]: Can we change the name to be a bit more specific? Something along the lines of `LibertyCXFPropertietTest`?

File: dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/PropertySettingTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+@RunWith(FATRunner.class)
+public class PropertySettingTest {
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/PropertySettingTest.java]: Same thing with the other test, can we change the name to be a bit more specific? Something along the lines of LibertyCXFPropertiesTest?

**Review by neuwerk — COMMENTED** [review]
Just a couple of name changes, other than that looks good.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+/*
+ * Tests the enablement of cxf.multipart.attachment property
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: Can you add some basic background to the structure of the test, where the test code came from aka copied what the `WebServiceRefTest` app, and then describe the flow of the test. Debugging tests in our bucket is always a little harder thanks to the tests we inherited not having good documentation?

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    @Test
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: I missed this the first time around, but you should describe in more detail what the property does and why it was added, that way if we have to ever debug this test we don't have to dig through CXF code changes to understand what's going on.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java (hunk 3 above)
Comment (neuwerk) [reply dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: Also specify that this test class should be used for any properties we use to override behavior in CXF.

File: dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/LibertyCXFPropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+@RunWith(FATRunner.class)
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/LibertyCXFPropertiesTest.java]: Same request as above, describe the flow of the test, where the code came from, and that this class is for overriding CXF behavior with properties.

File: dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/LibertyCXFPropertiesTest.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+		if (server1 != null) {
+			server1.stopServer();
+		}
+		if (client != null) {
+			client.stopServer();
+		}
+	}
+
+	/**
+	 * Client with URI policy attachment, service with URI policy attachment
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.X.wsat_fat/fat/src/com/ibm/ws/jaxws/wsat/fat/LibertyCXFPropertiesTest.java]: Same as above test method comment, can you describe in detail what the property we're testing does and where in CXF that change is located.

**Review by neuwerk — CHANGES_REQUESTED** [review]
The rename is great, but I missed some points about documentation in my first review. Thanks!!

**Comment by BerksanAtes** [comment]
#build

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipForRepeat;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+
+/*
+ * Tests the enablement of cxf.multipart.attachment
+ * and cxf.ignore.unsupported.policy property migrations
+ * from CXF 2.6.2. Test designed to work for EE9
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: I'd just mention that it's designed to work with both EE8 and EE9.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * This test is using WebServiceRefFeaturesTest's infrastructure
+ * with additional codes and wsdls tailored for this test only
+ */
+@RunWith(FATRunner.class)
+@SkipForRepeat(SkipForRepeat.NO_MODIFICATION)
+public class LibertyCXFPropertiesTest {
+
+    @Server("PropertyTestServer")
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: I'd change the name of the server to `LibertyCXFPropertiesServer` to continue one the better practices of our test buckets that match the server name with the fat test name, making the test easier to find when debugging failures.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, "webServiceRefFeatures", "com.ibm.ws.test.client.stub",
+                                      "com.ibm.ws.test.wsfeatures.client",
+                                      "com.ibm.ws.test.wsfeatures.client.handler",
+                                      "com.ibm.ws.test.wsfeatures.handler",
+                                      "com.ibm.ws.test.wsfeatures.service");
+
+        server.startServer("PropertyTestServer.log");
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: If you update the server name to `LibertyCXFPropertiesServer` you'll need to update this log file to `LibertyCXFPropertiesServer.log`

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java (hunk 4 above)
Comment (neuwerk) [reply dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/fat/src/com/ibm/ws/jaxws/fat/LibertyCXFPropertiesTest.java]: I'd still add some comments on the specifics of the test and what it does

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/publish/servers/PropertyTestServer/bootstrap.properties [hunk 10]
```diff
@@ -0,0 +1,5 @@
+bootstrap.include=../testports.properties
```
Comment (sawadood) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/publish/servers/PropertyTestServer/bootstrap.properties]: You could also define the properties in jvm.options file.

**Review by sawadood — APPROVED** [review]

**Comment by BerksanAtes** [comment]
#build

**Comment by BerksanAtes** [comment]
#build

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/build.gradle [hunk 11]
```diff
@@ -44,7 +44,7 @@ dependencies {
   'org.apache.cxf.xjc-utils:cxf-xjc-runtime:2.6.0',
   'org.apache.neethi:neethi:3.1.1',
   project(':com.ibm.ws.org.slf4j.api'),
-  project(':com.ibm.ws.org.slf4j.jdk14'),
+//  project(':com.ibm.ws.org.slf4j.jdk14'), // Uncomment before committing!
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/build.gradle]: Man, I have to do this too to run this bucket locally, but looks like you missed uncommenting it.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/test-applications/webServiceRefFeatures/src/com/ibm/ws/test/client/stub/ImageServicePropertyImplService.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/test-applications/webServiceRefFeatures/src/com/ibm/ws/test/client/stub/ImageServicePropertyImplService.java]: Copyright year is wrong for a new year.  FAT bucket didn't get run with it though.

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/test-applications/webServiceRefFeatures/src/com/ibm/ws/test/wsfeatures/client/WSAPropertyTestServlet.java [hunk 13]
```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/test-applications/webServiceRefFeatures/src/com/ibm/ws/test/wsfeatures/client/WSAPropertyTestServlet.java]: Same

**Review by neuwerk — COMMENTED** [review]
Just a couple of items to clean up! Good work @BerksanAtes

File: dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/build.gradle (hunk 11 above)
Comment (BerksanAtes) [reply dev/com.ibm.ws.jaxws.2.2.webcontainer_fat/build.gradle]: yeah I put a comment for myself too

**Review by neuwerk — APPROVED** [review]
Looks good! Thanks for all the hard work on this!

**Comment by BerksanAtes** [comment]
#run-libby-bot

**Pull request merged**